### DISCUSSION TITLE
onedrive client doesn't update the config

### DISCUSSION THREAD

**Discussion by mdehm** [abraunegg/onedrive#2974] [Q&A] [answered]
I have tried to create a custom config in order to synchronize a SharePoint folder, follwing:
[How to configure OneDrive SharePoint Shared Library sync
](https://github.com/abraunegg/onedrive/blob/master/docs/sharepoint-libraries.md)

I have created a custom folder for the sharepoint configuration at `~/.config/SharePoint`,
set the drive ID and the sync directory. 

However, when I try to update the onedrive config using
`onedrive --confdir="~/.config/SharePoint" --display-config` and display the result,
it still shows the standard onedrive config and not the updated one. See few lines
of the terminal output below. I have also tested to set `enable_logging` to true,
which is not regognized either. So it seems like I cannot update the onedrive config
following the guide mentioned above.

Do I overlook something?

```
Application version                          = onedrive v2.5.2-1+np1+1.1
Compiled with                                = LDC 2098
User Application Config path                 = /home/<user>/.config/SharePoint
System Application Config path               = /etc/onedrive
Applicable Application 'config' location     = /home/<user>/.config/SharePoint/config
Configuration file found in config location  = true - using 'config' file values to override application defaults
Applicable 'sync_list' location              = /home/<user>/.config/SharePoint/sync_list
Applicable 'items.sqlite3' location          = /home/<user>/.config/SharePoint/items.sqlite3
Config option 'drive_id'                     = 
Config option 'sync_dir'                     = ~/OneDrive
```
Note that `drive_id` is empty here and `sync_dir` is not updated correctly.

**Comment by abraunegg** [comment, marked as answer]
@martindehmel94 
You are 100% most likely doing *something* incorrect.

#### Example 'config' file
![image](https://github.com/user-attachments/assets/38c0cd46-a51a-4b3f-b48a-5c750fb3e772)


#### Application output when using --display-config
![image](https://github.com/user-attachments/assets/6cc7698b-8447-4757-933d-b80f42db806a)

Please *double check* how you have *editied* and/or created your configuration file and their entries.

> [!IMPORTANT]
> You are using Ubuntu, or a distribution based on Ubuntu 22.04.
> Please ensure you *read* https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#compatibility-with-curl very carefully and implement the HTTP/2 downgrade because of the significantly broken curl version your platform provides.
